src/measure_theory/integral/exp_decay.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2022 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+
+import measure_theory.integral.interval_integral
+
+import analysis.special_functions.exponential
+import analysis.special_functions.integrals
+import measure_theory.integral.integral_eq_improper
+
+
+/-!
+# Integrals with exponential decay at ∞
+
+As easy special cases of general theorems in the library, we prove the following test
+for integrability:
+
+* `exp_decay.integrable_bigoh_exp`: If `f` is continuous on `[a,∞)`, for some `a ∈ ℝ`, and there
+  exists `b > 0` such that `f(x) = O(exp(-b x))` as `x → ∞`, then `f` is integrable on `(a, ∞)`.
+
+-/
+
+noncomputable theory
+open real interval_integral measure_theory set filter
+
+namespace exp_decay
+
+/-- `exp(-b x)` is integrable on `(a, X]` for any finite `a`, `X`. -/
+lemma exp_neg_finite_integrable (b : ℝ) (a : ℝ) (X : ℝ):
+  integrable_on (λ x : ℝ, exp(-b * x) ) (Ioc a X) :=
+  (continuous_const.mul continuous_id).exp.integrable_on_Ioc
+
+/-- The antiderivative of `exp(-b x)` is what it should be. -/
+lemma exp_neg_hasderiv (b : ℝ) (x : ℝ) (h : 0 < b) :
+  has_deriv_at (λ (y : ℝ), -exp (-b * y) / b) (exp (-b * x)) x :=
+begin
+  have : has_deriv_at (λ (x : ℝ), x) 1 x := has_deriv_at_id x,
+  have : has_deriv_at (λ (x : ℝ), b*x) b x := by simpa only [mul_one] using
+    has_deriv_at.const_mul b this,
+  have : has_deriv_at (λ (x : ℝ), -b*x) (-b) x := by simpa only [neg_mul] using this.neg,
+  have : has_deriv_at (λ (x : ℝ), exp (-b*x)) (exp (-b*x)*(-b)) x := has_deriv_at.exp this,
+  have : has_deriv_at (λ (x : ℝ), -exp (-b*x)) (-(exp (-b*x)*(-b))) x := has_deriv_at.neg this,
+  have := (has_deriv_at.div_const this b),
+  have s : (-(exp (-b * x) * -b) / b) = exp(-b*x),
+  { simp only [neg_mul, mul_neg, neg_neg],
+    rw mul_div_cancel _ (h.ne') },
+  rw s at this, exact this
```

Comment:
@loefflerd By taking advantage of so-called dot-notation, you can chain most of the steps in the first part of the proof together: `(this.exp.neg.div_const b)`.
```suggestion
  have : has_deriv_at (λ (y : ℝ), -b*y) (-b) x,
  by simpa only [mul_one, neg_mul] using ((has_deriv_at_id x).const_mul b).neg,
  have : has_deriv_at (λ (y : ℝ), -exp (-b * y) / b) (-(exp (-b * x) * -b) / b) x :=
    (this.exp.neg.div_const b),
  simpa only [neg_mul, mul_neg, neg_neg, mul_div_cancel _ (h.ne')] using this,
```
The proof could be compressed even further, but I don't know if that is desirable.